packets.go

```diff
@@ -305,16 +314,49 @@ func (mc *mysqlConn) writeHandshakeResponsePacket(authResp []byte, plugin string
 	var authRespLEIBuf [9]byte
 	authRespLen := len(authResp)
 	authRespLEI := appendLengthEncodedInteger(authRespLEIBuf[:0], uint64(authRespLen))
-	if len(authRespLEI) > 1 {
+	if len(authRespLEI) > 1 && clientFlags&clientSecureConn != 0 {
```

Comment:
Would you show reference that describe this check is correct?

Reply:
Oh, clientSecureConn is set always. You wrote totally unnecessary check.
Please don't add code and increase complexity only for your preference.
Focus on implementing connectAttrs.

Reply:
You are right, thanks for checking @methane.
Fixed in https://github.com/go-sql-driver/mysql/pull/1241/commits/1058830325cb4ecdcea958eb5410661cab187a4f